sdk/servicebus/service-bus/test/streamingReceiver.spec.ts

```diff
@@ -1133,3 +1133,176 @@ describe("Streaming - Not receive messages after receiver is closed", function()
     await testReceiveMessages();
   });
 });
+
+describe("Streaming - onDetached", function(): void {
+  this.afterEach(async () => {
+    await afterEachTest();
+  });
+
+  it("doesn't call user's error handler on non-retryable errors", async function(): Promise<void> {
+    /**
+     * If onDetached is called with a non-retryable error, it is assumed that
+     * the onSessionError or onAmqpError has already called the user's
+     * error handler.
+     */
+    // Create the sender and receiver.
+    await beforeEachTest(
+      TestClientType.UnpartitionedQueue,
+      TestClientType.UnpartitionedQueue,
+      ReceiveMode.receiveAndDelete
+    );
+    // Send a message so we can be sure when the receiver is open and active.
+    await sender.send(TestMessage.getSample());
+    const receivedErrors: any[] = [];
+
+    let receiverIsActiveResolver: Function;
+    const receiverIsActive = new Promise((resolve) => {
+      receiverIsActiveResolver = resolve;
+    });
+    // Start the receiver.
+    receiver.registerMessageHandler(
+      async () => {
+        // Since we've received a message, mark the receiver as active.
+        receiverIsActiveResolver();
+      },
+      (err) => {
+        receivedErrors.push(err);
+      }
+    );
+
+    // Wait until we're sure the receiver is open and receiving messages.
+    await receiverIsActive;
+
+    // Simulate onDetached being called with a non-retryable error.
+    const nonRetryableError = translate(new Error(`I break systems.`));
+    nonRetryableError.retryable = false;
+    await (receiver as any)._context.streamingReceiver.onDetached(nonRetryableError);
```

Comment:
I think you can get a little bit more checking with:
receiver['_context']

it even works with autocomplete.

Reply:
Good tip, I didn't realize you could access `private` fields this way! @bterlson I'm just curious, is this technique something we can rely on to access private fields?

Reply:
@bterlson  - please say 'yes' because...I'm using it in places.

Reply:
Yep, fine for test code.

Reply:
I mean, fine for any time you would otherwise cast to any to reach into a private field. Fine for test code in the sense that hopefully you only want to do that in test code...

Reply:
Oh, it's definitely only for test code - this is one of those "private for others, modifiable so we can stub, etc..."